active/NumberBox/NumberBox.md

```diff
@@ -0,0 +1,220 @@
+the submission process.  When it's complete
+it will be incorporated into the public documentation at
+docs.microsoft.com (http://docs.microsoft.com/uwp/toolkits/winui/).
+Hopefully we'll be able to copy it mostly verbatim.
+So the second audience is everyone that reads there to learn how
+and why to use this API. -->
+
+## Background
+<!-- Use this section to provide background context for the new API(s) 
+in this spec. -->
+
+<!-- This section and the appendix are the only sections that likely
+do not get copied to docs.microsoft.com; they're just an aid to reading this spec. -->
+
+<!-- If you're modifying an existing API, included a link here to the
+existing page(s) -->
+
+<!-- For example, this section is a place to explain why you're adding this API rather than
+modifying an existing API. -->
+
+<!-- For example, this is a place to provide a brief explanation of some dependent
+area, just explanation enough to understand this new API, rather than telling
+the reader "go read 100 pages of background information posted at ...". -->
+
+
+## Description
+<!-- Use this section to provide a brief description of the feature.
+For an example, see the introduction to the PasswordBox control 
+(http://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box). -->
+
+Number box is a numerics only text control with support for input formatting, validation, basic calculation, and increment stepping.
+
+**Important APIs:** [NumberBox class](https://docs.microsoft.com/en-us/uwp/api/microsoft.ui.xaml.controls.numberbox)
+
+## Is this the right control? 
+
+Use a **NumberBox** control to recieve and display formatted numeric values such as currency, IP addresses, and telephone numbers. A **NumberBox** can also be enabled with basic calculator support to compute multiplication, division, addition, and subtraction across order of operations on entered equations. 
+
+## Examples
+<!-- Use this section to explain the features of the API, showing
+example code with each description. The general format is: 
+  feature explanation,
+  example code
+  feature explanation,
+  example code
+  etc.-->
+  
+<!-- Code samples should be in C# and/or C++/WinRT -->
+
+<!-- As an example of this section, see the Examples section for the PasswordBox control 
+(https://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box#examples). -->
+
+### Create a NumberBox
+
+XAML
+```XAML
+<NumberBox AutomationProperties.Name="simple NumberBox"/>
+```
+
+### Add placeholder text and a header
+
+XAML
+```XAML
+<TextBlock AutomationProperties.Name="simple NumberBox" Header="Enter equation:" PlaceholderText="A + B * C" />
+```
+
+### Enable calculation support
+
+XAML
+```XAML
+<TextBlock AutomationProperties.Name="NumberBox for simple calculations" 
+    Header="Enter equation:" 
+    PlaceholderText="A + B * C" 
+    AcceptsCalculation="True" />
+```
+
+### Add formatting
+
+```XAML
+<TextBlock AutomationProperties.Name="NumberBox for IP addresss" 
```

Comment:
If a more general [MaskedTextBox](https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.maskedtextbox?view=netframework-4.8) control is added later, will that obsolete this NumberBox feature? Should CustomFormatString align with MaskedTextBox.Mask?

Reply:
Masks could be added to the standard TextBox control, and as well as the MaskFormat, there should also be Suffix and Prefix

![image](https://user-images.githubusercontent.com/7389110/58600129-313a3f80-827b-11e9-908e-0ee3c6017ebc.png)

Reply:
@mdtauk If your proposal is that the prefix/suffix suggestion is to be added to the TextBox, then it would be better to create a separate issue for that idea. As it is stated that the NumberBox will inherit from TextBox then that change will be picked up by this control too. 
In general it's better to create new issues for new ideas, especially when they're about different controls. Having an discussion here about that idea risks confusing the discussion that's specific to the NumberBox spec (which is what this is.) It also enables that idea/discussion to be seen as a separate thing and not overlooked among discussion elsewhere.
For the record, I can see benefits about your suggestion but have lots of questions too.

Reply:
@mrlacey I was just responding to the comment prior, I have now added an at mention to make it clearer who it was directed to.

Reply:
I brought this up here because I don't want to look at this control in isolation. If we have a NumberBox now, and a MaskedTextBox later, we should think now about how they're not going to collide with each other in a confusing way.

Reply:
Resolving for now as @mdtauk has opened an issue for adding Prefix/Suffix to TextBox here:

https://github.com/microsoft/microsoft-ui-xaml/issues/784

Reply:
> Resolving for now as @mdtauk has opened an issue for adding Prefix/Suffix to TextBox here:
> 
> [microsoft/microsoft-ui-xaml#784](https://github.com/microsoft/microsoft-ui-xaml/issues/784)

FWIW Support for Masked Formatting is separate from Prefix and Suffix properties.